matlab/src/feather_reader.cc

```diff
@@ -102,34 +111,93 @@ mxArray* ReadVariableData(const std::shared_ptr<arrow::Column>& column) {
   return nullptr;
 }
 
-// Read the nulls of variable i from the Feather file as a mxArray*.
-mxArray* ReadVariableNulls(const std::shared_ptr<arrow::Column>& column) {
-  // TODO: Implement proper null value support. For the time being,
-  // we will simply return a zero initialized logical array to MATLAB.
-  return mxCreateLogicalMatrix(column->length(), 1);
+// arrow::Buffers are bit-packed, while mxLogical arrays aren't. This utility
+// uses arrow::BitUtil::GetBit to copy each bit of an arrow::Buffer into
+// each byte of an mxLogical array.
+void BitUnpackBuffer(const std::shared_ptr<arrow::Buffer>& source, const int64_t length,
+                     bool* destination) {
```

Comment:
We used `bool*` here since each `mxLogical` (true/false) value is represented by a full (non bit-packed) byte. 

Our intention was to improve readability by consistently representing the non bit-packed boolean values using `bool*`, and the bit-packed boolean values using `uint8_t*`. However we can also change this to using `uint8_t*` if it is more consistent with the Arrow C++ style.

Reply:
C++ does not guarantee that `bool` is byte-sized. It could be word-sized. So `uint8_t` should be used here if byte-sized values are intended.

Reply:
That's a good point. Since the size of `bool` is implementation defined, I [changed the code](https://github.com/apache/arrow/pull/4328/commits/ef3288da858dcfdb096ba743e6fb5e9d356280ef) to use `uint8_t*` instead of `bool*` here.

One issue I'm seeing after changing the bit-packing code to use `uint8_t` is that the bit-packing code is noticeably slower (by around 5-10% on Linux), although I'm not sure why this is happening. @pitrou Do you have any ideas about why this might happen?

Another way to resolve this issue is that we can use the [`mxLogical`](https://www.mathworks.com/help/matlab/apiref/mxlogical.html) type instead of `bool` or `uint8_t` here. This may be the appropriate datatype since [this part of the code](https://github.com/apache/arrow/blob/ef3288da858dcfdb096ba743e6fb5e9d356280ef/matlab/src/feather_reader.cc#L161) is writing the bits into an `mxLogical` matrix anyway.

Reply:
If that's the destination data type then that probably makes sense

Reply:
Thanks Wes, I have changed the bit-packing/unpacking code in `feather_reader.cc` and `feather_writer.cc` to use `mxLogical` instead of `uint8_t`.